src/main/java/nextstep/subway/path/domain/PathType.java

```diff
@@ -0,0 +1,20 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.line.domain.LineSection;
+
+public enum PathType {
+  DISTANCE {
+    @Override
+    public int getEdgeWeight(LineSection lineSection) {
+      return lineSection.getDistance();
+    }
+  },
+  DURATION {
+    @Override
+    public int getEdgeWeight(LineSection lineSection) {
+      return lineSection.getDuration();
+    }
+  };
+
+  public abstract int getEdgeWeight(LineSection lineSection);
```

Comment:
Function 을 사용하도록 해보면 어떨까요? 🤔 

```suggestion
  public int getEdgeWeight(LineSection lineSection) {
    return func.apply(lineSection);
  }
```